### PR TITLE
OOP school library: Set Up Associations

### PR DESCRIPTION
**Pull request by elarhadu**
In this milestone, the Project requirements were to;
- [x] Create a class Classroom with the following:
   - @label instance variable, should be initialized in the constructor.
   - Setter and getter for @label (remember about attr_accessor).
- [x] Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
   - Create the has-many side (a classroom has many students).
   - Create the belongs-to side (a student belongs to a classroom).
   - Make sure that when adding a student to a classroom it also sets the classroom for the student.
   - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- [x] Create a class Book with the following:
   - @title and @author instance variables, should be initialized in the constructor.
   - Setters and getters for instance variables (remember about attr_accessor).
- [x] Create a class Rental with the following:
   - @date instance variable, should be initialized in the constructor.
   - Setter and getter for @date (remember about attr_accessor).
- [x] Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
   - Create the has-many side of Book and Rental (a book has many rentals).
   - Create the belongs-to side of Rental and Book (a rental belongs to a book).
   - Create the has-many side of Person and Rental (a person has many rentals).
   - Create the belongs-to side of Rental and Person (a rental belongs to a person).
   - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -9,4 +9,9 @@ def initialize(age, classroom, name = 'Unknown', parent_permission: true)
   def play_hooky
     '¯\\_(ツ)_/¯'
   end
+
+  def add_classroom(classroom)
+    @classroom = classroom
+    classroom.students << self
```
Comment (Kamba56) [inline student.rb]: - [Optional] Consider checking if the student already exists in the classroom before pushing. Use 
`classroom.students.push(self) unless classroom.students.include?(self)`.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :author, :title, :rentals
+
```
Comment (Kamba56) [inline book.rb:3]: - [ ] Kindly create a many-to-many relationship between Person and Book class through rentals. Add a method in the Book class for adding rentals. Also, add a method for adding rentals in the Person class. Check the example [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship).

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @elarhadu, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great use of Github workflow!👏👏👏
- [x] All linters passed
- [x] Great job setting up associations between students and classrooms

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 2 above)
Comment (elarhadu) [reply book.rb:3]: Thank you @Kamba56  for pointing this out. I'll rectify it immediately.

File: student.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply student.rb]: Fixed.

File: book.rb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply book.rb:3]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @elarhadu 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**